File: feature-translate/src/main/java/com/andyanika/translator/feature/translate/di/TranslationModule.java
package com.andyanika.translator.feature.translate.di;

import android.arch.lifecycle.ViewModel;

import com.andyanika.resources.ViewModelKey;
import com.andyanika.translator.common.scopes.FragmentScope;
import com.andyanika.translator.feature.translate.TranslationViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import dagger.multibindings.IntoMap;
import io.reactivex.subjects.PublishSubject;

@Module
abstract class TranslationModule {
    @Provides
    static PublishSubject<String> provideRetrySubject() {
        return PublishSubject.create();
    }

    @Binds
    @IntoMap
    @FragmentScope
    @ViewModelKey(TranslationViewModel.class)
    abstract ViewModel bindViewModel(TranslationViewModel viewModel);
}

File: feature-translate/src/main/java/com/andyanika/translator/feature/translate/di/TranslationModule.kt
package com.andyanika.translator.feature.translate.di

import com.andyanika.translator.common.scopes.FragmentScope
import com.andyanika.translator.feature.translate.TranslationFragment
import com.andyanika.translator.feature.translate.TranslationView
import dagger.Binds
import dagger.Module
import dagger.Provides
import io.reactivex.rxjava3.subjects.PublishSubject

@Module
internal interface TranslationModule {
    @Binds
    @FragmentScope
    fun getView(fragment: TranslationFragment): TranslationView

    companion object {
        @Provides
        fun provideRetrySubject(): PublishSubject<String> {
            return PublishSubject.create()
        }
    }
}
